app/serializers/base_bookmarkable.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+##
+# Anything that we want to be able to bookmark must be registered as a
+# bookmarkable type using Bookmark.register_bookmarkable(bookmarkable_klass),
+# where the bookmarkable_klass is a class that implements this BaseBookmarkable
+# interface. Some examples are TopicBookmarkable and PostBookmarkable.
+#
+# These methods are then called by the Bookmarkable class through a public
+# interface, used in places where we need to list, send reminders for,
+# or otherwise interact with bookmarks in a way that is unique to the
+# bookmarkable type.
+class BaseBookmarkable
```

Comment:
I think this should be in `app/services` instead of `app/serializers`?

Reply:
Ah yeah will fix